library/firewall_lib.py

```diff
@@ -315,7 +315,9 @@ def main():
                 ],
             ),
             state=dict(
-                choices=["enabled", "disabled", "present", "absent"], required=True
+                choices=["enabled", "disabled", "present", "absent"],
+                required=False,
+                default=None,
```

Comment:
I think the default should still be `"enabled"` - this is the general convention in Ansible

Reply:
hmm - on second thought - adding a default value would be a breaking change to the API - so I think what you have here is correct

Reply:
Yeah, I had thought about that, but arrived at the same conclusion.